crates/sui-core/src/authority/authority_store.rs

```diff
@@ -738,6 +738,32 @@ impl AuthorityStore {
         Ok(())
     }
 
+    /// Insert objects directly into the object table, but do not touch other tables.
+    /// This is used in fullnode to insert objects from validators certificate handling response
+    /// in fast path execution.
+    pub(crate) fn insert_objects_to_object_store(&self, objects: &Vec<Object>) -> SuiResult {
+        let mut write_batch = self.perpetual_tables.objects.batch();
```

Comment:
this method doesn't update reference counters properly

Reply:
logic of keeping consistent indirect object representation is a little bit complicated so if possible let's at least filter them out here
you can do:
```
     let StoreObjectPair(store_object, indirect_object) =
          get_store_object_pair(obj.clone(), self.indirect_objects_threshold);

      if indirect_object.is_some() {
          continue;
      }
      write_batch.insert_batch(
          &self.perpetual_tables.objects,
          std::iter::once((ObjectKey(obj.id(), obj.version()), store_object)),
      )?;
```

Reply:
fixed, ty!